### PR TITLE
Replacement for the new queue

### PR DESCRIPTION
**Pull request by kaste**
Now, the strategy is just to use the Python standard library.

### CODE REVIEW

**Comment by kaste** [comment]
I still forgot something ... 🤣

**Comment by braver** [comment]
Good busy! 😄

**Comment by kaste** [comment]
Okidoki. 

Let's have true parallel linting. 🔥 🎆

**Comment by kaste** [comment]
We're several seconds (!) faster at startup now.

**Comment by braver** [comment]
Wow, it *is* fast now. Excellent!

**Comment by braver** [comment]
@pykong or @FichteFoll want to say/ask anything smart about this before I pull it?

**Comment by FichteFoll** [comment]
The diff is kinda hard to read without all the surrounding architectural changes (or how it's been before in the first place).

I'll take a note to look at it properly in full when I have more time available next week.

**Comment by kaste** [comment]
Since this is a hotfix, I'll do a quick walkthrough.  The overall diff is kinda useless bc this is already a complete replacement of the queue from last week. We have two relevant commits here: https://github.com/SublimeLinter/SublimeLinter/pull/763/commits/a369376f85a5cedeef80ffdb2f042f9782337755 which introduces the new debouncer and https://github.com/SublimeLinter/SublimeLinter/pull/763/commits/1b1d5a21682aa988355e7cfe4159ea8d53d8fcec which finally introduces a lock to limit the parallelism.

The disadvantage of using `sublime.set_timeout_async` was that it runs on a *shared* thread. If we block here, all other events like `on_selection_modified_async` etc. will not run before we finish.

The implementation here thus follows the python standard library. We replace 

```
sublime.set_timeout_async(worker, delay * 1000)
```

directly with

```
threading.Timer(delay, worker).start()
```

This will already solve most of our problems. `threading.Timer` has a nice feature ` sublime.set_timeout_async` does not have. We can actually abort or cancel scheduled tasks. So most of the logic, we had before just disappears. Every time we 'hit', we just `cancel()` the previous task if any. 

After that, we basically can run `lint` in parallel. This is safe for different views which have each their own linters. So we just use a `threading.Lock` per `vid` in https://github.com/SublimeLinter/SublimeLinter/pull/763/commits/1b1d5a21682aa988355e7cfe4159ea8d53d8fcec

File: lint/queue.py [hunk 1]
```diff
@@ -1,12 +1,15 @@
-import sublime
-
 from . import persist
 
+from collections import defaultdict
 import time
+import threading
 
+# TODO: Both `timers` and `running` herein grow unbounded. T.i. a proper
```
Comment (braver) [inline lint/queue.py]: Shouldn't we do something about this now?

Comment (kaste) [reply lint/queue.py]: No, it's a good solution and a hotfix. Holding 100 or so locks should be no problem. Don't know. Is it?

It's one of the moving parts. Can we make the debouncer self-contained, so the debouncer decides when to clear up old timers, or do we need to trigger the clearance from outside? 

This async/parallel stuff just takes a week to sink. This is really hard to think about. At least for me.

Comment (kaste) [reply lint/queue.py]: Okay, should I just do it? I add it to `def on_pre_close(self, view)`?

Comment (braver) [reply lint/queue.py]: > This async/parallel stuff just takes a week to sink. This is really hard to think about. At least for me.

Definitely.

**Comment by braver** [comment]
To me this seems to work fine, a lot more performant than what we had first. I’m going to merge this so it spreads to more users. We can always fix or discuss or do whatever to improve it.

File: lint/queue.py (hunk 1 above)
Comment (braver) [reply lint/queue.py]: > Okay, should I just do it?

Add the cleanup to `on_pre_close` you mean? Doesn't sound like a bad idea.

Comment (kaste) [reply lint/queue.py]: Ah, let's do it

**Pull request merged**